### PR TITLE
Oryx Pro Switchable Graphics View

### PR DESCRIPTION
**Pull request by mmstick**
screenshot from 2018-05-24 11-07-22

### CODE REVIEW

File: gnome-initial-setup/pages/summary/gis-summary-page.c [hunk 1]
```diff
@@ -336,6 +363,34 @@ gis_summary_page_constructed (GObject *object)
 
   G_OBJECT_CLASS (gis_summary_page_parent_class)->constructed (object);
 
+  if (has_switchable_graphics ()) {
+    char *left_desc = _("Use the system menu on the top panel to switch between "
+      "Intel and NVIDIA graphics.  Switching will prompt you to restart your "
+      "device.");
+
+    char *right_desc = _("To increase battery life, your Oryx Pro defaults to "
+      "Intel graphics.  To use external displays, switch to NVIDIA graphics.");
```
Comment (cassidyjames) [inline gnome-initial-setup/pages/summary/gis-summary-page.c]: Minor nitpick: Sentences should be separated by a single space.

File: gnome-initial-setup/pages/summary/gis-summary-page.c [hunk 2]
```diff
@@ -336,6 +363,34 @@ gis_summary_page_constructed (GObject *object)
 
   G_OBJECT_CLASS (gis_summary_page_parent_class)->constructed (object);
 
+  if (has_switchable_graphics ()) {
+    char *left_desc = _("Use the system menu on the top panel to switch between "
+      "Intel and NVIDIA graphics.  Switching will prompt you to restart your "
```
Comment (cassidyjames) [inline gnome-initial-setup/pages/summary/gis-summary-page.c]: Minor nitpick: Sentences should be separated by a single space.

**Comment by mmstick** [comment]
I can dynamically insert the appropriate product name. Power is down at Carson though, so it may be a while before I can commit.

**Comment by mmstick** [comment]
I implemented the stricter DMI check, too. So `oryp4` and `oryp4-b` are currently whitelisted.

**Comment by jackpot51** [comment]
The build is failing with this message:

```
15:39:51 The following files contain translations and are currently not in use. Please
15:39:51 consider adding these to the POTFILES.in file, located in the po/ directory.
15:39:51 
15:39:51 gnome-initial-setup/pages/summary/gis-summary-page-oryx-switchable.ui
15:39:51 
15:39:51 If some of these files are left out on purpose then please add them to
15:39:51 POTFILES.skip instead of POTFILES.in. A file 'missing' containing this list
15:39:51 of left out files has been written in the current directory.
15:39:51 if [ -r missing -o -r notexist ]; then \
15:39:51   exit 1; \
15:39:51 fi
```

File: gnome-initial-setup/pages/summary/gis-summary-page.c [hunk 3]
```diff
@@ -328,6 +334,109 @@ update_distro_name (GisSummaryPage *page)
...
+    has_switchable = strcmp (trimmed, "oryp4") == 0
+      || strcmp (trimmed, "oryp4-b") == 0;
+    free (product_version);
+  }
+
+  return has_switchable;
+}
+
+static char*
+get_product_name ()
```
Comment (jackpot51) [inline gnome-initial-setup/pages/summary/gis-summary-page.c]: You are not using this function

File: gnome-initial-setup/pages/summary/gis-summary-page.c [hunk 4]
```diff
@@ -328,6 +334,109 @@ update_distro_name (GisSummaryPage *page)
...
+}
+
+static void
+gis_summary_page_set_switchable_descriptions (GisSummaryPagePrivate *priv)
+{
+  char *left_desc = _("Use the system menu on the top panel to switch between "
+    "Intel and NVIDIA graphics. Switching will prompt you to restart your "
+    "device.");
+
+  char *right_desc = _("To increase battery life, your Oryx Pro defaults to "
```
Comment (jackpot51) [inline gnome-initial-setup/pages/summary/gis-summary-page.c]: You need to use the product name here as well

**Pull request merged**